lib/src/components/NxTextInput/NxTextInput.tsx

```diff
@@ -48,6 +49,7 @@ export const PrivateNxTextInput = forwardRef<HTMLDivElement, Props>(
       const isTextArea = type === 'textarea',
           element = isTextArea ? 'textarea' : 'input',
           typeAttr = isTextArea ? undefined : (type || 'text'),
+          classes = rows && isTextArea ? 'nx-text-input__input textarea-custom-sizing' : 'nx-text-input__input',
```

Comment:
I think this should be managed within `NxCodeSnippet`.  That's where the styling for this class is defined, and additionally we aren't trying to add behavior to `NxTextInput`, just `NxCodeSnippet`.